extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,437 @@
+			this.$container = $('#notification-container');
+			this.$markAllAsReadButton = $('#mark-all-as-read-button');
+
+			this.registerEvents = function () {
+				this.addDropdownLoadingEvent();
+				this.addMarkAllAsReadEvent();
+			};
+
+			this.addMarkAllAsReadEvent = function () {
+				this.$markAllAsReadButton.click($.proxy(this.logic.markAllAsRead, this.logic));
+			};
+
+			this.addDropdownLoadingEvent = function () {
+				var $dropdown = $('#on-site-notifications-dropdown');
+				$dropdown.click(function () {
+					logic.loadFirstPage();
+				});
+			};
+
+			this._mapToView = function (notifications) {
+				function getIcon(type) {
+					if (type === notificationTypes.discussionReply) {
+						return 'wds-icons-reply-small';
+					} else if (type === notificationTypes.announcement) {
+						return 'wds-icons-megaphone';
+					} else {
+						return 'wds-icons-upvote-small';
+					}
+				}
+
+				function getAvatars(actors) {
+					return actors.slice(0, 5);
+				}
+
+				return notifications.map(this.proxy(function (notification) {
+					return {
+						icon: getIcon(notification.type),
+						uri: notification.uri,
+						snippet: notification.snippet,
+						text: this.textFormatter.getText(notification),
+						isUnread: notification.isUnread,
+						communityName: notification.communityName,
+						showAvatars: notification.totalUniqueActors > 2 && notification.type === notificationTypes.discussionReply,
+						showAvatarOverflow: notification.totalUniqueActors > 5,
+						avatarOverflow: notification.totalUniqueActors - 5,
+						avatars: getAvatars(notification.latestActors)
+					}
+				}));
+			};
+
+			this.renderNotifications = function (notifications) {
+				var html = mustache.render(this.template, this._mapToView(notifications));
+				this.$container.append(html);
+
+				const markAsRead = this.proxy(function (e) {
+					try {
+						const id = $(e.target).closest('.wds-notification-card').attr('id');
+						this.logic.markAsRead(id);
+					} catch (e) {
+						log('Failed to mark as read ' + e, log.levels.error, logTag);
+					}
+					return false;
+				});
+
+				$(this.$container).find('.wds-notification-card__icon-wrapper').click(markAsRead);
+			};
+
+			this.renderUnreadCount = function (count) {
+				this.unreadCount = count;
+
+				if (this.unreadCount > 0) {
+					this.$markAllAsReadButton.addClass('wds-is-visible');
+					this.$notificationsCount.html(this.unreadCount).parent('.bubbles').addClass('show');
+				} else {
+					this.$markAllAsReadButton.removeClass('wds-is-visible');
+					this.$notificationsCount.empty().parent('.bubbles').removeClass('show');
+				}
+			};
+
+			/**
+			 * .classRemove does not work on SVG
+			 * @param element
+			 */
+			function removeIsUnreadClass(element) {
+				element.classList.remove('wds-is-unread');
+			}
+
+			function findUnreadAndClearClass($element) {
+				$element.find('.wds-icon.wds-is-unread').each(function (_, e) {
```

Comment:
Per [jQuery docs](https://api.jquery.com/each/) it's an [Element](http://api.jquery.com/Types/#Element). The same value is stored in `this`.